tests/framework/extensions/prime/primechecks.go

```diff
@@ -0,0 +1,86 @@
+package primechecks
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	client "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/rancherversion"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	PodResourceSteveType = "pod"
+	rancherImage         = "rancher"
+)
+
+// CheckUIBrand checks the UI brand of Rancher Prime. If the Rancher instance is not Rancher Prime, the UI brand should be blank.
+func CheckUIBrand(client *rancher.Client, isPrime bool, brand *client.Setting, rancherBrand string) string {
+	if isPrime && rancherBrand == brand.Value {
+		logrus.Infof("Rancher Prime UI brand: %s", brand.Value)
+		return brand.Value
+	}
+
+	return brand.Value
+}
+
+// CheckVersion checks the if Rancher Prime is set to true and the version of Rancher.
+func CheckVersion(isPrime bool, rancherVersion string, serverConfig *rancherversion.Config) string {
+	if isPrime && rancherVersion == serverConfig.RancherVersion {
+		logrus.Infof("Rancher Prime: %t | Version: %s", isPrime, serverConfig.RancherVersion)
+		return serverConfig.RancherVersion
```

Comment:
what's the difference between this return value and the one at line 37? I just seems like it will log if the condition is met, and that's it.

Reply:
Wouldn't it make sense that if it doesn't satisfy the condition that you return an empty string, or a custom error like `fmt.Errorf("version fails prime check")` or something like that? Because imo this function doesn't really add much besides logging.

Reply:
I will add error messages in separate conditions. This refactoring mainly was to help clean up the `prime_test.go`, so that is why logging is really only thing in these functions, to add context.

Reply:
Added error messages based on new conditions.